File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/OnOfflineRouteFoundCallback.java
package com.mapbox.services.android.navigation.v5.navigation;

import android.support.annotation.NonNull;

import com.mapbox.api.directions.v5.models.DirectionsRoute;

/**
 * Callback used for finding offline routes.
 */
public interface OnOfflineRouteFoundCallback {

  /**
   * Called when an offline route is found.
   *
   * @param route offline route
   */
  void onRouteFound(@NonNull DirectionsRoute route);

  /**
   * Called when there was an error fetching the offline route.
   *
   * @param error with message explanation
   */
  void onError(@NonNull OfflineError error);
}

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/OnOfflineRouteFoundCallback.kt
package com.mapbox.services.android.navigation.v5.navigation

import com.mapbox.api.directions.v5.models.DirectionsRoute

/**
 * Callback used for finding offline routes.
 */
interface OnOfflineRouteFoundCallback {

    /**
     * Called when an offline route is found.
     *
     * @param route offline route
     */
    fun onRouteFound(route: DirectionsRoute)

    /**
     * Called when there was an error fetching the offline route.
     *
     * @param error with message explanation
     */
    fun onError(error: OfflineError)
}
